cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java

```diff
@@ -0,0 +1,263 @@
+      sparkProperties.put("spark.cassandra.auth.password", getEnv("CASSANDRA_PASSWORD", ""));
+    }
+
+    /** When set, this indicates which jars to distribute to the cluster. */
+    public Builder jars(String... jars) {
+      this.jars = jars;
+      return this;
+    }
+
+    /** Keyspace to store dependency rowsToLinks. Defaults to "zipkin2" */
+    public Builder keyspace(String keyspace) {
+      this.keyspace = checkNotNull(keyspace, "keyspace");
+      return this;
+    }
+
+    /** Day (in epoch milliseconds) to process dependencies for. Defaults to today. */
+    public Builder day(long day) {
+      this.day = midnightUTC(day);
+      return this;
+    }
+
+    /** Ensures that logging is setup. Particularly important when in cluster mode. */
+    public Builder logInitializer(Runnable logInitializer) {
+      this.logInitializer = checkNotNull(logInitializer, "logInitializer");
+      return this;
+    }
+
+    /** Comma separated list of hosts / IPs part of Cassandra cluster. Defaults to localhost */
+    public Builder contactPoints(String contactPoints) {
+      this.contactPoints = contactPoints;
+      return this;
+    }
+
+    /** The local DC to connect to (other nodes will be ignored) */
+    public Builder localDc(@Nullable String localDc) {
+      this.localDc = localDc;
+      return this;
+    }
+
+    public CassandraDependenciesJob build() {
+      return new CassandraDependenciesJob(this);
+    }
+  }
+
+  final String keyspace;
+  final long day;
+  final String dateStamp;
+  final SparkConf conf;
+  @Nullable final Runnable logInitializer;
+
+  CassandraDependenciesJob(Builder builder) {
+    this.keyspace = builder.keyspace;
+    this.day = builder.day;
+    SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+    df.setTimeZone(TimeZone.getTimeZone("UTC"));
+    this.dateStamp = df.format(new Date(builder.day));
+    this.conf = new SparkConf(true).setMaster(builder.sparkMaster).setAppName(getClass().getName());
+    conf.set("spark.cassandra.connection.host", parseHosts(builder.contactPoints));
+    conf.set("spark.cassandra.connection.port", parsePort(builder.contactPoints));
+    if (builder.localDc != null) conf.set("connection.local_dc", builder.localDc);
+    if (builder.jars != null) conf.setJars(builder.jars);
+    for (Map.Entry<String, String> entry : builder.sparkProperties.entrySet()) {
+      conf.set(entry.getKey(), entry.getValue());
+    }
+    this.logInitializer = builder.logInitializer;
+  }
+
+  public void run() {
+    long microsLower = day * 1000;
+    long microsUpper = (day * 1000) + TimeUnit.DAYS.toMicros(1) - 1;
+
+    log.info(
+        "Running Dependencies job for {}: {} ≤ Span.timestamp {}",
+        dateStamp,
+        microsLower,
+        microsUpper);
+
+    SparkContext sc = new SparkContext(conf);
+    try {
+      JavaRDD<DependencyLink> links = javaFunctions(sc)
+          .cassandraTable(keyspace, "span")
```

Comment:
TODO: suppose we could limit this to data stored an hour before and 5m after the window (ex to decode and filter less data)

Reply:
Seem like a really good performance improvement for folks with lots of data.  The current spark job doesn't run in a realistic amount of time on our current dataset.  Limiting to a subset of the data would be a big win.  Probably an improvement for a later time though?